src/Control/Effect/Resumable.hs

```diff
@@ -99,10 +99,10 @@ runResumableWithC :: (forall x . err x -> m x) -> ResumableWithC err m a -> m a
 runResumableWithC f (ResumableWithC m) = m f
 
 instance (Carrier sig m, Monad m) => Carrier (Resumable err :+: sig) (ResumableWithC err m) where
-  gen a = ResumableWithC (\ _ -> gen a)
-  alg = algR \/ algOther
-    where algR (Resumable err k) = ResumableWithC (\ f -> f err >>= runResumableWithC f . k)
-          algOther op = ResumableWithC (\ f -> alg (handlePure (runResumableWithC f) op))
+  ret a = ResumableWithC (const (ret a))
+  eff op = ResumableWithC (\ handler -> (alg handler \/ eff . handlePure (runResumableWithC handler)) op)
+    where alg :: Monad m => (forall x . err x -> m x) -> Resumable err (ResumableWithC err m) (ResumableWithC err m a) -> m a
+          alg handler (Resumable err k) = handler err >>= runResumableWithC handler . k
```

Comment:
This algebra needs a signature now since it’s parameterized by the rank-_n_ quantified error handler.